File: solutions/zig/space-age/1/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const sec_f64: f64 = @as(f64, @floatFromInt(seconds));

        const earth_year_seconds: f64 = 31557600.0;

        const mul: f64 = switch (self) {
            .mercury => 0.2408467,
            .venus => 0.61519726,
            .earth => 1,
            .mars => 1.8808158,
            .jupiter => 11.862615,
            .saturn => 29.447498,
            .uranus => 84.016846,
            .neptune => 164.79132,
        };

        return (sec_f64 / earth_year_seconds) / mul;
    }
};

File: solutions/go/space-age/1/space_age.go
package spaceage

type Planet string

func getPlanetPeriod(planet Planet) float64 {
	switch planet {
	case "Mercury":
		return 0.2408467
	case "Venus":
		return 0.61519726
	case "Earth":
		return 1.0
	case "Mars":
		return 1.8808158
	case "Jupiter":
		return 11.862615
	case "Uranus":
		return 84.016846
	case "Saturn":
		return 29.447498
	case "Neptune":
		return 164.79132
	default:
		return -1.0
	}
}

func Age(seconds float64, planet Planet) float64 {
	period := getPlanetPeriod(planet)
	if period > 0 {
		return seconds / getPlanetPeriod(planet) / 31557600
	}

	return -1.0
}
